packages/client/src/controllers/auth-controller.ts

```diff
@@ -102,3 +102,21 @@ export const approveUser = (clientId: string) => {
   });
   console.log('ApproveUser sent');
 };
+
+export const denyUser = (clientId: string) => {
+  const body = makeApproveOrDenyJoinMessage(false, clientId);
+
+  reqResponseService.send(body, response => {
+    if (response === 'timeout') {
+      // TODO display the error in the gui
```

Comment:
Trivia: This "show in gui" was mostly relevant to an older concept for user join. I wanted to put a long timeout on the "joinWhiteboard" message, and sent a success/error from the server only after host's approval. So a timeout on join was pretty important, since it basically meant denied join.
But the solution with having a separate state in the client is definitely more elegant. Which also means that  this is less important for display to the user, unless you implement it anyway as part of the wider popus implementation.

Reply:
Neat, thanks for the info! I decided to leave the `TODO`s as they're very ugly and make me sad, which fuels my want to actually create something better. In other words, pop-ups next!